model/Image.php

```diff
@@ -328,6 +342,97 @@ public function generateSetSize(Image_Backend $backend, $width, $height) {
 		return $backend->paddedResize($width, $height);
 	}
 
+	/**
+	 * Proportionally scale down this image if it is wider or taller than the specified dimensions.
+	 * Similar to SetRatioSize but without up-sampling. Use in templates with $ScaleMax.
+	 * @see Image::SetRatioSize()
+	 *
+	 * @param integer $width The maximum width of the output image
+	 * @param integer $height The maximum height of the output image
+	 * @return Image
+	 */
+	public function ScaleMax($width, $height) {
+		if ($this->width > $width || $this->height > $height) return $this->SetRatioSize($width,$height);
+		return $this;
+	}
+
+	/**
+	 * Proportionally scale down this image if it is wider than the specified width. 
+	 * Similar to SetWidth but without up-sampling. Use in templates with $ScaleMaxWidth.
+	 * @see Image::SetWidth()
+	 *
+	 * @param integer $width The maximum width of the output image
+	 * @return Image
+	 */
+	public function ScaleMaxWidth($width) {
+		if ($this->width > $width) return $this->SetWidth($width);
+		return $this;
+	}
+
+	/**
+	 * Proportionally scale down this image if it is taller than the specified height. 
+	 * Similar to SetHeight but without up-sampling. Use in templates with $ScaleMaxHeight.
+	 * @see Image::SetHeight()
+	 *
+	 * @param integer $height The maximum height of the output image
+	 * @return Image
+	 */
+	public function ScaleMaxHeight($height) {
+		if ($this->height > $height) return $this->SetHeight($height);
+		return $this;
+	}
+
+	/**
+	 * Crop this image to the aspect ratio defined by the specified width and height, 
+	 * then scale down the image to those dimensions if it exceeds them.
+	 * Similar to CroppedImage but without up-sampling. Use in templates with $CropMax.
+	 * @see Image::CroppedImage()
+	 *
+	 * @param integer $width The relative (used to determine aspect ratio) and maximum width of the output image
+	 * @param integer $height The relative (used to determine aspect ratio) and maximum height of the output image
+	 * @return Image
+	 */
+	public function CropMax($width, $height) {
+		// Is the image already the correct size?
+		if ($this->isSize($width, $height)) return $this;
+		
+		// If not, make sure the image isn't upsampled
+		$imageRatio = $this->width / $this->height;
```

Comment:
In case the file isn't found, $this->height will be zero, so it could divide by zero here if you're not careful. :D